control-plane/agents/src/bin/core/pool/pool_operations.rs

```diff
@@ -30,6 +30,11 @@ impl ResourceLifecycle for OperationGuardArc<PoolSpec> {
                 node_id: request.node.to_string(),
             });
         }
+        if request.disks.len() != 1 {
```

Comment:
Do we need this check in the control plane?

Reply:
Ok I think it's fine to have it for now, though we might need to change this if we ever have different pool types.